packages/translations/src/index.ts

```diff
@@ -22,22 +22,25 @@ import type {
  * @returns boolean which indicates if the languages are loaded
  */
 export function useTranslations(initParams: TranslationInitParams): boolean {
-  const [initialized, setInitialized] = useState(i.isInitialized)
+  const [initialized, setInitialized] = useState(false)
+  const [loaded, setLoaded] = useState(false)
 
   !initialized && initTranslations(initParams)
 
   useEffect(() => {
-    const handleInitialized = () => setInitialized(true)
-
-    // Check if initialized between render and useEffect
-    if (i.isInitialized) return handleInitialized()
+    const handleInitialized = () => {
+      setInitialized(true)
+    }
+    const handleLoaded = () => {
+      setLoaded(true)
+    }
 
     // Listen if initialization takes longer
     i.on('initialized', handleInitialized)
-    return () => i.off('initialized', handleInitialized)
```

Comment:
Is there a reason you no longer unsubscribe from the event?

Reply:
refactored it now so it uses a callback instead of the event listeners

Reply:
If this hook is used in a short-lived component (or strict mode), React may warn that a state setter was called after the component unmounted.

One option to handle this is to ignore the setter if the component was unmounted before the translations were loaded:
```tsx
  useEffect(() => {
    let mounted = true
    initTranslations({
      onDone: () => {
        mounted && setLoaded(true)
      },
      ...initParams,
    })
    return () => {
      mounted = false
    }
  })
```

Reply:
good catch, adding it